src/components/Map/algorithms/Chart.tsx

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import {
+  Chart as ChartJS,
+  CategoryScale,
+  LinearScale,
+  PointElement,
+  LineElement,
+  Title,
+  Tooltip,
+  Filler,
+  Legend,
+} from 'chart.js';
+import { Line } from 'react-chartjs-2';
+
+ChartJS.register(
+  CategoryScale,
+  LinearScale,
+  PointElement,
+  LineElement,
+  Title,
+  Tooltip,
+  Filler,
+  Legend
+);
+
+export const options = {
+  responsive: true,
+  plugins: {
+    legend: {
+      position: 'top' as const,
+    },
+    title: {
+      display: true,
+      text: 'Diffusion Centrality per node over time',
+    },
+  },
+};
+
+
+function convertToDataFormat() {
+  // data: Map<string, Map<string, Map<string, number>>>
+  const diffcen = {
+  '1': 
+    {"u": {"w": 0.875, "u": 0.22285507026263682, "v": 0.125},
+    "w": {"v": 0.8333333333333333, "w": 1.899891188788434, "u": 0.16666666666666666}, 
+    "v": {"v": 1.0028478161818657, "u": 0.027777777777777776, "w": 0.9722222222222222}},
+  '5': 
+    {"w": {"u": 0.11447557860207201, "v": 0.5253881911125526, "w": 7.548832973261137}, 
+    "u": {"u": 1.1971055926702734, "v": 0.3869554744587899, "w": 0.5306180750216268}, 
+    "v": {"u": 0.08736238508669167, "v": 5.302357097157161, "w": 0.5498104456545143}}, 
+  '3': 
+    {"u": {"w": 0.5846156302145143, "u": 0.7100239287446046, "v": 0.3458195569108133}, 
+    "w": {"u": 0.12356102280699693, "v": 0.5789979029092013, "w": 3.3594015098012524}, 
+    "v": {"v": 3.15260229690905, "u": 0.07788491452768159, "w": 0.6169780536897971}}, 
+  '4': 
+    {"w": {"v": 0.4474437258633965, "u": 0.10132558606505494, "w": 4.382201150288006}, 
+    "u": {"u": 0.9657539481298845, "v": 0.44933065647196085, "w": 0.459774563353038}, 
+    "v": {"u": 0.10096360051946618, "v": 4.298012880591784, "w": 0.45620812795185256}}, 
+  '2': 
+    {"v": {"w": 0.4560563664334602, "u": 0.1013954341203142, "v": 2.1497846458869967}, 
+    "u": {"w": 0.46385720089360516, "v": 0.4551294396765634, "u": 0.4789370417123615}, 
+    "w": {"v": 0.4474339020518443, "w": 2.191212980781382, "u": 0.10138603482247471}}
+  } 
+  const labels = Object.keys(diffcen);
+
+  const node_to_time_values = new Map<string, number[]>();
+
+  for (const [time, time_map] of Object.entries(diffcen)) {
```

Comment:
Object.entries(dictionary) is the only way this works that I could come up with.